app/questionnaire/questionnaire_store_updater.py

```diff
@@ -95,21 +89,27 @@ def remove_list_item_and_answers(self, list_name: str, list_item_id: str):
             list_item_id=list_item_id
         )
 
-    def add_completed_location(self, location=None, section_id=None):
+        self._progress_store.remove_progress_for_list_item_id(list_item_id=list_item_id)
+
+    def add_completed_location(self, location=None):
         location = location or self._current_location
-        section_id = section_id or self._current_section_id
 
-        self._progress_store.add_completed_location(section_id, location)
+        self._progress_store.add_completed_location(location)
 
     def remove_completed_location(self, location=None):
         location = location or self._current_location
-        self._progress_store.remove_completed_location(
-            self._current_section_id, location
-        )
+        self._progress_store.remove_completed_location(location)
+
+    def update_section_status(self, section_status, location=None):
+        if location:
+            section_id = location.section_id
+            list_item_id = location.list_item_id
+        else:
+            section_id = self._current_location.section_id
+            list_item_id = self._current_location.list_item_id
 
-    def update_section_status(self, section_status):
         self._progress_store.update_section_status(
-            self._current_section_id, section_status
+            section_status, section_id=section_id, list_item_id=list_item_id
```

Comment:
Should the arguments of this match the questionnaire_store method?

Reply:
Maybe, felt weird to pass in location to update a section status, so got questionnaire store update to handle it. Could name it `update_section_status_for_location`?

Reply:
Have changed it to match the progress store method.